src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -212,27 +199,39 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
             'URL'
         };
 
+        // these fields have been whitelisted since they have special handling in BDI
+        Set<String> allowedFields = new Set<String>{
+            'DonationCampaignImported__c'
+        };
+
         // looping through the fields on DataImport__c and remove unsupported data types
-        for (String fieldName : sortedFieldNames) {
-            DescribeFieldResult dfr = fieldMap.get(fieldName);
+        for (Schema.DescribeFieldResult dfr : fieldMap.values()) {
             String dataImportFieldApiName = UTIL_Namespace.StrTokenRemoveNSPrefix(dfr.getName());
 
             String dataImportFieldMappedObject = BDI_DataImportService.getTargetObject(dfr.getName());
-            Boolean fieldAllowed = allowedObjects.contains(dataImportFieldMappedObject)
+            Boolean fieldAllowed = (allowedObjects.contains(dataImportFieldMappedObject)
                 && allowedDataTypes.contains(dfr.getType().name())
                 && !bannedFields.contains(dataImportFieldApiName)
-                && dfr.getInlineHelpText() != null;
+                && dfr.getInlineHelpText() != null)
+                || allowedFields.contains(dataImportFieldApiName);
 
             if (fieldAllowed) {
-                dfrs.add(dfr);
+                dfrMap.put(dfr.label, dfr);
             }
         }
 
-        BGE_BatchGiftEntry_UTIL.checkFieldPermissions(dfrs);
+        BGE_BatchGiftEntry_UTIL.checkFieldPermissions(dfrMap.values());
 
-        return dfrs;
-    }
+        List<String> sortedFieldLabels = new List<String>(dfrMap.keySet());
+        sortedFieldLabels.sort();
+
+        List<Schema.DescribeFieldResult> sortedDFRs = new List<Schema.DescribeFieldResult>();
+        for (String fieldLabel : sortedFieldLabels) {
+            sortedDFRs.add(dfrMap.get(fieldLabel));
+        }
 
+        return sortedDFRs;
```

Comment:
Moved this down here from above so that we can sort on Field Label instead of Field API Name.